core/sql/executor/ExHbaseSelect.cpp

```diff
@@ -444,6 +444,7 @@ ExWorkProcRetcode ExHbaseScanSQTaskTcb::work(short &rc)
   Lng32 retcode = 0;
   rc = 0;
   Lng32 remainingInBatch = batchSize_;
+  NABoolean isFirstBatch = false;
```

Comment:
It depends on the queue size. Which is set by the compiler and can be modified by the executor. But just as a general design principle, work methods are designed to interrupt their work at any time (often due to some resource constraint) so it is best to use members for any state that is needed for the life of the operator.